boat_utilities/src/boat_utilities/angles.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+# NOTE: All operations in this module use degrees CCW from East, as in a unit circle
+# TODO: Switch to rads
+__UNITS = 360.0
+
+
+def cosd(angle):
+    """Calculate the cosine of the following angle, in degrees."""
+    from math import cos, radians
+    return cos(radians(angle))
+
+
+def sind(angle):
+    """Calculate the sine of the following angle, in degrees."""
+    from math import sin, radians
+    return sin(radians(angle))
+
+
+def tand(angle):
+    """Calculate the tangent of the following angle, in degrees."""
+    from math import tan, radians
+    return tan(radians(angle))
+
+
+def atan2d(y, x):
+    """Calculate the arctan of the following angle, in degrees."""
+    from math import atan2, degrees
+    return degrees(atan2(y, x))
+
+
+def normalize(angle):
+    """Normalize the angle to be 0 to 360."""
+    from math import fmod
+    return fmod(fmod(angle, __UNITS) + __UNITS, __UNITS)
+
+
+def normalize_signed(angle):
+    """Normalize the angle to be -180 to 180."""
+    angle = normalize(angle)
+    if angle > __UNITS/2:
+        angle -= __UNITS
+    return angle
+
+
+def is_within_bounds(val, lower, upper):
+    """Determine whether lower < val < upper.
+
+    Args:
+        val The value to check
+        lower The lower bound
+        upper The upper bound
+    Returns:
+        `True` if the value is between the specified bounds
+    """
+    val = normalize(val)
+    lower = normalize(lower)
+    upper = normalize(upper)
+
+    if val > upper:
+        val -= __UNITS
+    if lower > upper:
+        lower -= __UNITS
+
+    return lower <= val and val <= upper
+
+
+def is_on_right(angle, ref):
```

Comment:
What happens if both `is_on_right` and `is_on_left` return false because its exactly 180 degrees away? I think maybe one of these should have an equality so we catch all cases.

Reply:
Good catch. I wrote this under the assumption that all our values would be floating point, and so I was a bit sloppy. I'm not sure what the best solution is here though. I don't really like the idea of making these asymmetrical.
 - Perhaps we could add an argument (default False) that sets the func to inclusive instead of exclusive and leaves it to the caller to handle it. That just moves the issue up the stack though.
 - Maybe we just cast everything to a float and let it randomly pick one side or the other. That's kinda gross though.

I'm not sure what the right solution here is. Thoughts?

Reply:
Hmm I remember when I wrote my dirty versions I just had the `is_on_right` absorb the equality but I agree that asymmetry is dirty.  I like that first solution, I'm just afraid someone will use it without considering that possibility... So I think either we do that, and put a comment to be careful, or there should be a utility that does both and returns 1 for right and -1 for left (defined as those directions somewhere as an enum or whatever) and that function sets the argument to make sure that a developer doesn't forget to specify it.